SimCalorimetry/EcalSimAlgos/src/EcalSignalGenerator.cc

```diff
@@ -41,17 +41,57 @@ CaloSamples EcalSignalGenerator<EBDigitizerTraits>::samplesInPE(const DIGI &digi
 
   CaloSamples result(detId, digi.size());
 
+  // correction factor for premixed sample: ratio of laser corrections
+  float correction_factor_for_premixed_sample_transparency = 1.0;
```

Comment:
@amassiro , is there any sense making variable name for 50 characters? Another concerns to this code: a) if there is a mixture of double and float numbers, usually efficiency of the code is lower than if all are double or all are float. It is possible to save memory if use float values in containers but here I do not see any; b) there are too many commented lines which better to remove.

Reply:
@amassiro 
Do you follow up on this?

Reply:
Is it a problem in having a variable with a long name? In coding I prefer to have things very readable, so that future people reading the code can see what is happening.

I will move the new variable(s) I defined to double (if it is more efficient to have all double instead of double and float, I did not know), but the rest of the code that uses double is out of my control, since it's out of my control since it was like that since 20 years. 
I suppose a general review of "how much things are affected if we move from double to float" could provide interesting results and possible improvements, but it does not refer specifically to this PR.

About the commented lines, do you refer to "code commented"? If yes, I'm sorry, I thought I fixed them, after Kevin comments.
If they are pure comments, they are meant to help a reader understand the code, for the years to come.

@srimanob : there is something that has to be decided centrally: how will the time dependent generation will be performed? Do we have already a test workflow?
I have tested the ECAL time dependent part privately, but having a central workflow (for example defining the run ranges, ...) will help in both the development and the conditions preparation.

Reply:
@Dr15Jones  
Could you please circulate on what do you think the workflow configuration should look like?

@amassiro 
I think if the code part is ready, we can start by relvals which we can adapt any change in configuration after the release is available.

FYI @fabiocos